libraries/Microsoft.Bot.Builder.LanguageGeneration/EvaluationOptions.cs

```diff
@@ -0,0 +1,126 @@
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Bot.Builder.LanguageGeneration
+{
+    /// <summary>
+    /// Options for LG rendering the line break in text.
+    /// </summary>
+    public enum LGLineBreakStyle
+    {
+        /// <summary>
+        /// Default mode.
+        /// </summary>
+        Default,
+
+        /// <summary>
+        /// Markdown mode.
+        /// </summary>
+        Markdown,
+    }
+
+    /// <summary>
+    /// Options for evaluation of LG template <see cref="EvaluationOptions"/> class.
+    /// </summary>
+    public class EvaluationOptions
+    {
+        private static readonly Regex NullKeyReplaceStrRegex = new Regex(@"\${\s*path\s*}");
+        private readonly string strictModeKey = "@strict";
+        private readonly string replaceNullKey = "@replaceNull";
+        private readonly string lineBreakKey = "@lineBreakStyle";
+
+        public EvaluationOptions()
+        {
+            this.StrictMode = null;
+            this.NullSubstitution = null;
+            this.LineBreakStyle = null;
+        }
+
+        public EvaluationOptions(EvaluationOptions opt)
+        {
+            this.StrictMode = opt.StrictMode;
+            this.NullSubstitution = opt.NullSubstitution;
+            this.LineBreakStyle = opt.LineBreakStyle;
+        }
+
+        public EvaluationOptions(IList<string> optionStrList)
+        {
+            if (optionStrList != null)
+            {
+                foreach (var optionStr in optionStrList)
+                {
+                    if (!string.IsNullOrWhiteSpace(optionStr) && optionStr.Contains("="))
+                    {
+                        var index = optionStr.IndexOf('=');
+                        var key = optionStr.Substring(0, index).Trim();
+                        var value = optionStr.Substring(index + 1).Trim().ToLower();
+                        if (key == strictModeKey)
+                        {
+                            if (value == "true")
+                            {
+                                StrictMode = true;
+                            }
+                        }
+                        else if (key == replaceNullKey)
+                        {
+                            NullSubstitution = (path) => NullKeyReplaceStrRegex.Replace(value, $"{path}");
+                        }
+                        else if (key == lineBreakKey)
+                        {
+                            LineBreakStyle = value.ToLower() == LGLineBreakStyle.Markdown.ToString().ToLower() ? LGLineBreakStyle.Markdown : LGLineBreakStyle.Default;
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the option of rendering new line characters.
+        /// </summary>
+        /// <value>
+        /// if lineBreakStyle is 'markdown', a new line character will be replaced by two new line characters. Otherwise, it will be the same.
```

Comment:
>if lineBreakStyle is 'markdown', a new line character will be replaced by two new line characters. Otherwise, it will be the same. [](start = 12, length = 130)

Usually value is just a description of the value and behavior would be in remarks.  Also initial cap.

Reply:
changed the description